### PR TITLE
Port yuzu-emu/yuzu#2457: "yuzu/{about_dialog, main}: Specify string conversions explicitly for SCM-related info"

### PR DESCRIPTION
**Pull request by FearlessTobi**
See yuzu-emu/yuzu#2457 for more info.

**Original description**:
Specifies the conversions explicitly to avoid implicit conversions from const char* to QString. This makes it easier to disable implicit QString conversions in the future.

In this case, the implicit conversion was technically wrong as well. The implicit conversion treats the input strings as ASCII characters. This would result in an incorrect conversion being performed in the rare case a branch name was created with a non-ASCII Unicode character, likely resulting in junk being displayed.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4774)

### CODE REVIEW

**Comment by wwylele** [comment]
I am not sure I fully understood the second commit, as the commit description doesn't match the diff. Also apparently it changes the title style which might needs a screenshot in the PR description.

**Comment by zhaowenlan1779** [comment]
What's the status of this?

Regarding the title bar style, I'd say we should likely keep the current style which is more user-friendly IMO.

**Comment by FearlessTobi** [comment]
Okay, reverted the status bar change.

File: src/citra_qt/aboutdialog.cpp [hunk 1]
```diff
@@ -11,10 +11,10 @@ AboutDialog::AboutDialog(QWidget* parent)
     : QDialog(parent, Qt::WindowTitleHint | Qt::WindowCloseButtonHint | Qt::WindowSystemMenuHint),
       ui(new Ui::AboutDialog) {
     ui->setupUi(this);
-    ui->labelLogo->setPixmap(QIcon::fromTheme("citra").pixmap(200));
-    ui->labelBuildInfo->setText(
-        ui->labelBuildInfo->text().arg(Common::g_build_fullname, Common::g_scm_branch,
-                                       Common::g_scm_desc, QString(Common::g_build_date).left(10)));
+    ui->labelLogo->setPixmap(QIcon::fromTheme(QStringLiteral("Citra")).pixmap(200));
```
Comment (lioncash) [inline src/citra_qt/aboutdialog.cpp]: Was changing `citra` to `Citra` here intentional?

Comment (FearlessTobi) [reply src/citra_qt/aboutdialog.cpp]: no.

**Pull request merged**